File: MiniDOMOffice/CRM/Chat/Rooms/CChatRoomCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;
using static minidom.Office;

namespace minidom
{
    public partial class Messenger
    {

        /// <summary>
        /// Cursore sulla tabella delle stanze
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class CChatRoomCursor 
            : minidom.Databases.DBObjectCursor<CChatRoom>
        {
            
            private DBCursorStringField m_Name = new DBCursorStringField("Name");

            /// <summary>
            /// Costruttore
            /// </summary>
            public CChatRoomCursor()
            {
            }

            /// <summary>
            /// Name
            /// </summary>
            public DBCursorStringField Name
            {
                get
                {
                    return m_Name;
                }
            }

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Messenger.Rooms;
            }
             
        }
    }
}

File: MiniDOMOffice/CRM/Chat/Rooms/CChatRoomCursor.vb
Imports minidom
Imports minidom.Sistema
Imports minidom.Databases
Imports minidom.CustomerCalls

Partial Public Class Messenger

    ''' <summary>
    ''' Cursore sulla tabella delle stanze
    ''' </summary>
    ''' <remarks></remarks>
    Public Class CChatRoomCursor
        Inherits DBObjectCursor(Of CChatRoom)

        Private m_Name As New CCursorFieldObj(Of String)("Name")

        Public Sub New()
        End Sub

        Public ReadOnly Property Name As CCursorFieldObj(Of String)
            Get
                Return Me.m_Name
            End Get
        End Property

        Protected Overrides Function GetConnection() As CDBConnection
            Return minidom.Messenger.Database
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Messenger.Rooms.Module
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_ChatRooms"
        End Function
         

    End Class

End Class
 